### PR TITLE
Fix tag counts for dashboards and queries.

### PR DESCRIPTION
**Pull request by jezdez**
Refs #3108.

### CODE REVIEW

**Comment by jezdez** [comment]
FWIW, the main part is just b9f254a5b19dadfa4ab1b0792bc7086762d07758. Feel free to ignore the rest, which I only did to better understand the query.

File: docker-compose.yml [hunk 1]
```diff
@@ -39,5 +39,7 @@ services:
     # The following turns the DB into less durable, but gains significant performance improvements for the tests run (x3
     # improvement on my personal machine). We should consider moving this into a dedicated Docker Compose configuration for
     # tests.
+    ports:
+      - "15432:5432"
```
Comment (arikfr) [inline docker-compose.yml]: ☝️

File: redash/models.py [hunk 2]
```diff
@@ -1390,26 +1390,19 @@ def search(cls, org, groups_ids, user_id, search_term):
...
-        query = query.filter(or_(Dashboard.user_id == user.id, Dashboard.is_draft == False))
-
-        return query.order_by(usage_count.desc())
+            db.session
+            .query(tag_column, usage_count)
+            .group_by(tag_column)
+            .filter(Dashboard.id.in_(dashboards.options(load_only('id'))))
+            .order_by(usage_count.desc())
+        )
+        return query
```
Comment (arikfr) [inline redash/models.py]: Can I bother you with adding a test case for this bug & method?

File: docker-compose.yml (hunk 1 above)
Comment (jezdez) [reply docker-compose.yml]: Yeah, so question: I often do that to inspect the database during development using a macOS app ([Postico](https://eggerapps.at/postico/)). Do you mind having that exposed for development?

File: redash/models.py (hunk 2 above)
Comment (jezdez) [reply redash/models.py]: No bother :)

File: docker-compose.yml (hunk 1 above)
Comment (arikfr) [reply docker-compose.yml]: You could use Redash to inspect it ;-) But sure, I guess there is no harm in leaving this.

Comment (jezdez) [reply docker-compose.yml]: Heh, I do that, too, in some cases. But in others it's easier to just use other software, plus it allows editing as well (e.g. reverting archiving queries and dashboards).

**Comment by arikfr** [comment]
> The query looks like this btw:

Most of the complexity comes from the fact that permissions to dashboards are based on permissions to queries, which are based on permissions to data sources... In the not so far future I'm hoping to change this. Hopefully it will make the queries saner in the process.

**Comment by arikfr** [comment]
Thank you for updating the Query query too. Once we make the subquery a parameter `all_tags` receives, it looks like we can move it into a mixin class, which we can reuse for both Query and Dashboard.

Probably not worth tackling now, but worth making a mental note for the future.

**Comment by jezdez** [comment]
Yeah, I was about to do that, but then remembered that I've already split `models.py` in #3026 introducing a `redash/models/mixins.py`. Let's wait for this to land and then we can refactor this a tiny bit.

**Comment by arikfr** [comment]
Thanks!

**Pull request merged**